Inventory.md

```diff
@@ -0,0 +1,173 @@
+
+Genesis and slotting information should trivially be the same. 
+
+The fee policies work differently. 
+
+Jörmungandr charges `constant + coeff * (inputs + outputs) + optional_cert_specific_fee`
+
+The Haskell Byron node charges `constant + coeff * tx_size_in_bytes`
+
+some links:
+
+- [Wallet BlockchainParameters](https://github.com/input-output-hk/cardano-wallet/blob/468e0fb14ed3b50ab3d16ba127f24242a41ed126/lib/core/src/Cardano/Wallet/Primitive/Types.hs#L1084-L1102)
+- [cardano-ledger ProtocolParameters](https://github.com/input-output-hk/cardano-ledger/blob/0ad59ce46b3da8af6f6af97064e7c77cc7c2ac8b/cardano-ledger/src/Cardano/Chain/Update/ProtocolParameters.hs#L32-L62)
+- [wallet use of Jörmungandr fee policy](https://github.com/input-output-hk/cardano-wallet/blob/468e0fb14ed3b50ab3d16ba127f24242a41ed126/lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs#L105)
+- [doc of Jörmungandr fee policy: format.md](https://github.com/input-output-hk/chain-libs/blob/master/chain-impl-mockchain/doc/format.md)
+- [cardano-ledger TxFeePolicy](https://github.com/input-output-hk/cardano-ledger/blob/0ad59ce46b3da8af6f6af97064e7c77cc7c2ac8b/cardano-ledger/src/Cardano/Chain/Common/TxFeePolicy.hs#L41-L59)
+
+
+### Protocol Parameter Updates
+
+Jörmungandr does not fully support protocol parameter updates. If protocol parameters can change, the wallet needs to be aware.
+
+- Jörmungandr: static (updates not implemented)
+- Haskell Byron: static (updates possible, but not likely)
+- Haskell Shelley: dynamic (updates possible, wallet should handle this)
+
+### Blocks
+
+The Jörmungandr wallet uses the `BlockHeader = (slotId, blockHeight, hash, prevHash)` record to identify points on the chain.
+
+The Byron Haskell node only *needs* `Point ByronBlock = Origin | At (slotNo, hash)` to start giving us blocks. The wallet can retrieve `blockHeight` and `prevHash` from block headers. But those two fields are not needed to initiate the chain-sync protocol. Small ❌. 
+
+We might need to convert the genesis point `Origin`, to a `BlockHeader` if we are to persist it in the DB ❌
+
+Byron `SlotNo` counts slots from genesis in a single number. The current Jörmungandr-compatible wallet `SlotId` counts the epoch AND the slot in that epoch. ❌
+
+In Jörmungandr a slot can only be inhabited by a single block. On the Haskell side, epoch boundary blocks will share slot with the normal block that comes after it. ❌
+
+There is no limit on how far the Jörmungandr nodes can rollback. The Haskell Nodes have a limit: (`k` blocks). Haskell nodes have a clear separation between mutable and immutable parts of the chain.
+
+### Transactions
+
+Transaction inputs in Jörmungandr consist of (txHash, index, coinValue). In the Haskell Byron implementation they are only (txHash, index). ❌
+
+The addresses we get from the Haskell chain-sync protocol are not byte-strings, but rather in their deserialized form.
```

Comment:
I don't get this? Addresses can certainly be serialize if needed, right ?

Reply:
Yes. Still a difference. 

At a glance it seems wasteful to 1. decodeAddress, 2. encodeAddress, 3. decodeAddress. But it's easy to implement.

We could treat addresses more generically to avoid this.

(I say basically this in the discussion, on L106)

Reply:
It's hard to be more generic than `Bytestring` though :grimacing: ... We _could_ make address a type parameter of the wallet engine but.. trust me, no one wants to go through this.

Reply:
> It's hard to be more generic than Bytestring though

Quite the opposite though, right? 😬  With `ByteString` we can do whatever we want with it. `addr` requires us to be explicit with what we can do with it.

> trust me, no one wants to go through this

I'm at least curious / open to the possibility :)